### PR TITLE
RISC-V GNU/Linux as host platform

### PR DESCRIPTION
**Pull request by msizanoen1**
This PR add a new builder named `dist-riscv64-linux` that builds the compiler toolchain for RISC-V 64-bit GNU/Linux.

r? @alexcrichton

### CODE REVIEW

**Comment by msizanoen1** [comment]
@tblah

**Comment by tblah** [comment]
Looks good to me, thanks!

**Comment by alexcrichton** [comment]
Code-wise this looks ok, but I'll leave it up to @pietroalbini for the final approval to confirm the infra team has capacity for this.

r? @pietroalbini

File: src/ci/docker/dist-riscv64-linux/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
+FROM debian:10
```
Comment (pietroalbini) [inline src/ci/docker/dist-riscv64-linux/Dockerfile:1]: All of our images are based off Ubuntu, is there any reason you chose Debian here?

**Comment by pietroalbini** [comment]
This adds RISC-V as a Tier 2 platform, has the compiler team approved the change?

**Comment by malbarbo** [comment]
If the compiler team are considering this addition, I would like to ask to also consider the addition of android host platform, as requested in https://github.com/rust-lang/rust/issues/42639.

**Comment by msizanoen1** [comment]
It is already a tier 2 platform.

File: src/ci/docker/dist-riscv64-linux/Dockerfile (hunk 1 above)
Comment (msizanoen1) [reply src/ci/docker/dist-riscv64-linux/Dockerfile]: This is now changed to Ubuntu 20.04.

**Comment by msizanoen1** [comment]
https://github.com/rust-lang/rust-forge/pull/356

**Comment by pietroalbini** [comment]
> It is already a tier 2 platform.

Well, it's a tier 2 cross-compilation platform, but this PR moves it to a tier 2 host platform, which increases the amount of things we test on CI.

@nikomatsakis does the compiler team want to weight on moving RISC-V Linux to a tier 2 host platform (it's just a tier 2 cross-compilation platform now)?

**Comment by kinnison** [comment]
For reference, `rustup` is ready to provide a binary on this target thanks to the work of @tblah

**Comment by nikomatsakis** [comment]
I'm comfortable with adding new tier platforms if the infra support is there *and* if we setup a group of folks to notify for help when problems arise. I've been a bit slow in doing the work to create that group for windows, but I think we could readily create a group for RISC-V at the same time.

**Comment by nikomatsakis** [comment]
(I didn't mean that to be the "opinion of the @rust-lang/compiler team", though. We don't really have a clear set of rules for what it takes to move something from Tier to Tier. I have to review what we said in our design meeting to remember how we felt about this.)

**Comment by chaozju** [comment]
@msizanoen1 
Hi !

Is there a beta way to boot riscv host rust?

I tried on your [branch](https://github.com/msizanoen1/rust/tree/riscv-host) for native compiling in qemu but in some ways still failed

**Comment by pietroalbini** [comment]
In [today's infrastructure team meeting](https://zulip-archive.rust-lang.org/242791tinfra/1051820200617meeting.html#201146733) we approved allocating one builder for riscv64-unknown-linux-gnu. I'll defer to the compiler team for a final approval of the uplift to Host Tier 2.

**Comment by nikomatsakis** [comment]
@pietroalbini https://github.com/rust-lang/compiler-team/issues/312 has been approved

File: src/ci/docker/host-x86_64/dist-various-1/crosstool-ng.sh [hunk 2]
```diff
@@ -1,4 +1,3 @@
-#!/bin/bash
```
Comment (pietroalbini) [inline src/ci/docker/host-x86_64/dist-various-1/crosstool-ng.sh:1]: Why was this change made?

File: src/ci/azure-pipelines/auto.yml [hunk 3]
```diff
@@ -53,6 +53,7 @@ jobs:
       dist-powerpc-linux: {}
       dist-powerpc64-linux: {}
       dist-powerpc64le-linux: {}
+      dist-riscv64-linux: {}
```
Comment (pietroalbini) [inline src/ci/azure-pipelines/auto.yml:56]: You also need to add the builder to GHA.

**Review by emilyalbini — CHANGES_REQUESTED** [review]

File: src/ci/azure-pipelines/auto.yml (hunk 3 above)
Comment (msizanoen1) [reply src/ci/azure-pipelines/auto.yml:56]: Added.

File: src/ci/docker/host-x86_64/dist-various-1/crosstool-ng.sh (hunk 2 above)
Comment (msizanoen1) [reply src/ci/docker/host-x86_64/dist-various-1/crosstool-ng.sh:1]: It was an oversight during rebase. That file is only used for RISC-V builds and it should be moved out of that folder.

**Review by emilyalbini — APPROVED** [review]

**Comment by pietroalbini** [comment]
Thanks!

@bors r+ rollup=iffy

**Pull request merged**